### PR TITLE
Add unzip argument to download from url method

### PR DESCRIPTION
**Pull request by IvanPombo**
Based on the least astonishment principle and on giving control to the user, this PR adds an argument to unzip the file in `download_from_url` function. The previous behaviour extracted the argument automatically if the the file in the url had zip extension.

The existing test was modified to include both behaviour.

### CODE REVIEW

File: inductiva/utils/files.py [hunk 1]
```diff
@@ -130,7 +130,9 @@ def _unzip(zip_path: pathlib.Path, dest_path: pathlib.Path):
     zip_path.unlink()
 
 
-def download_from_url(url: str, save_dir: Optional[str] = None) -> str:
+def download_from_url(url: str,
+                      save_dir: Optional[str] = None,
+                      unzip: bool = False) -> str:
```
Comment (hpenedones) [inline inductiva/utils/files.py:135]: this is still a bit surprising for the user, I think... because the method can download any file, but there is an option that is called "unzip", which actually only applies to downloading zip files... What if I pass here "True" and it is not a zip file? I guess it will be ignored...

Comment (hpenedones) [reply inductiva/utils/files.py:135]: When we pass unzip = True for a zip file, what I would expect from the description is that file would be unzipped to a local folder, with all its contents. But in fact this only applies to zip files with one single file... and there is more: then that file will be renamed (not super clear to what -- the docstring mentions 'local_file_path' but that's unclear...)

Comment (hpenedones) [reply inductiva/utils/files.py:135]: My take: isn't this method trying to do too many things?
Feels like it should only download a file, that's it. Or if we pass an unzip option, it should not be surprising: just decompress it all, independently of number of files, and do not do any renaming.

Seems like this method is doing all this, because historically we wanted to use it in some snippets of documentation and we want the code there to be simple and compact, but that's not a strong reason enough, to create a method that is super "entangled" with many behaviours inside it... 
Probably we just need a simpler way to store example files of the API, but not implement a method with surprising behaviours. 
What do you think?

Comment (IvanPombo) [reply inductiva/utils/files.py:135]: I see your points! I will look into this and make it simpler and not so entangled.

File: inductiva/molecules/utils.py [hunk 2]
```diff
@@ -47,7 +46,7 @@ def align_trajectory_to_average(universe, trajectory_output_path):
                     in_memory=False).run()
 
 
-def download_pdb_from_rcsb(pdb_id: str, save_dir: Optional[str] = None) -> str:
+def download_pdb_from_rcsb(pdb_id: str) -> str:
```
Comment (begaiym-k) [inline inductiva/molecules/utils.py:49]: Why did you remove save_dir?

If so you need to remove it in the docstring as well.

File: inductiva/utils/files.py [hunk 3]
```diff
@@ -108,80 +107,41 @@ def get_alphanum_key(file):
...
-            zip_ref.extractall(dest_path.parent)
-
-    # Rename the file to dest_path
+        zip_ref.extractall()
+
     zip_path.unlink()
 
 
-def download_from_url(url: str, save_dir: Optional[str] = None) -> str:
+def download_from_url(url: str, unzip: bool = False) -> str:
```
Comment (begaiym-k) [inline inductiva/utils/files.py:119]: Just curious, Is there any reason why the users would not want to unzip it right away?

**Review by begaiym-k — APPROVED** [review]
Thanks @IvanPombo! I left couple of comments, other than that LGTM!

File: inductiva/molecules/utils.py (hunk 2 above)
Comment (IvanPombo) [reply inductiva/molecules/utils.py:49]: It will be by default, to not add extra complexity to the download.

File: inductiva/utils/files.py (hunk 3 above)
Comment (IvanPombo) [reply inductiva/utils/files.py:119]: This can be used for several things, and the unzip tells they users they are actually uncompressing.
